apps/newsletters-ui/src/app/components/NewsletterForm.tsx

```diff
@@ -0,0 +1,228 @@
+		if (name === '') return;
+
+		const revisedData = {
+			...newsletter,
+			...deriveNewsletterFieldsFromName(name),
+		};
+		updateDataAndWarnings(revisedData);
+	};
+
+	const updateDataAndWarnings = (
+		revisedData: Partial<Record<keyof Newsletter, unknown>>,
+	) => {
+		setWarnings({});
+		const parseResult = newsletterSchema.safeParse(revisedData);
+		const issueMap: Partial<Record<keyof Newsletter, string>> = {};
+		if (existingIds.includes(revisedData.identityName as string)) {
+			issueMap.identityName =
+				'There is an existing newsletters with this identityName';
+		}
+
+		if (parseResult.success) {
+			setNewsletter(parseResult.data);
+		} else {
+			parseResult.error.issues.forEach((issue) => {
+				const { message, path, code } = issue;
+				const key =
+					typeof path[0] === 'string'
+						? (path[0] as keyof Newsletter)
+						: undefined;
+
+				if (key) {
+					issueMap[key] = message || code;
+				}
+			});
+
+			const parseResultAllowingEmptyStrings =
+				newsletterSchemaAllowingEmptyStrings.safeParse(revisedData);
+			if (parseResultAllowingEmptyStrings.success) {
+				setNewsletter(parseResultAllowingEmptyStrings.data);
+			}
+		}
+		setWarnings(issueMap);
+	};
+
+	const handleSubmit = () => {
+		alert('not implemented');
+	};
+
+	return (
+		<div>
+			<h2>Create newsletter form</h2>
+			<button onClick={deriveValuesFromName} disabled={newsletter.name === ''}>
+				Derive values from name
+			</button>
+
+			<SchemaForm
+				schema={newsletterSchema}
+				data={newsletter}
+				changeValue={manageChange}
+				showUnsupported
+				options={{
+					regionFocus: ['UK', 'AUS', 'US'],
+				}}
+				excludedKeys={[
+					'emailEmbed',
+					'illustration',
+					'brazeSubscribeAttributeNameAlternate',
+				]}
+				readOnlyKeys={['listId']}
+				validationWarnings={warnings}
+			/>
+
+			<ArrayInput
+				label="brazeSubscribeAttributeNameAlternate"
+				validationWarning={warnings.brazeSubscribeAttributeNameAlternate}
+				data={newsletter.brazeSubscribeAttributeNameAlternate ?? []}
+				change={(data) => {
+					manageArrayChange(data, 'brazeSubscribeAttributeNameAlternate');
+				}}
+			/>
+
+			<fieldset>
+				<legend>emailEmbed</legend>
+				<SchemaForm
+					schema={emailEmbedSchema}
+					data={newsletter.emailEmbed}
+					changeValue={manageEmailEmbedChange}
+					showUnsupported
+					validationWarnings={{}}
+				/>
+			</fieldset>
+
+			{/* TO DO - controll to add the illustration field to the newsletter if undefined */}
+			{newsletter.illustration && (
+				<fieldset>
+					<legend>emailEmbed</legend>
+					<SchemaForm
+						schema={illustrationSchema}
+						data={newsletter.illustration}
+						changeValue={manageIllustrationChange}
+						showUnsupported
+						validationWarnings={{}}
+					/>
+				</fieldset>
+			)}
+
+			<button onClick={handleSubmit} disabled={hasWarnings}>
+				Create Newsletter
+			</button>
+
+			<NewsletterDetail newsletter={newsletter} />
+		</div>
+	);
+};
```

Comment:
This is a really long file - could some parts of it be split out to make it more manageable?

Reply:
I could split out the definition of the blank newsletter. 
Unfortunately, factoring code out of Functional Components is difficult as it tends to rely on hooks like `useState` which need to be defined within the function body. We probably could, try refactoring some of the code into custom hooks or higher-level functions, but I'm not sure how to do that without adding more complexity.
Its a good thought, though - It might be something to re-visit when we have some more implementations of the `SchemaForm` so we can see what abstractions or patterns would fit.